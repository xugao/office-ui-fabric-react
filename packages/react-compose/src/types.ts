import * as React from 'react';

// tslint:disable-next-line:interface-name
export interface ShorthandConfig<TProps> {
  mappedProp?: keyof TProps;
  mappedArrayProp?: keyof TProps;
  allowsJSX?: boolean;
}

//
// "as" type safety
//

export type PropsOfElement<
  // tslint:disable-next-line:no-any
  E extends keyof JSX.IntrinsicElements | React.JSXElementConstructor<any>
> = JSX.LibraryManagedAttributes<E, React.ComponentPropsWithRef<E>>;

// tslint:disable-next-line:interface-name
export interface ComponentWithAs<TElementType extends React.ElementType = 'div', TProps = {}>
  extends React.FunctionComponent {
  <TExtendedElementType extends React.ElementType = TElementType>(
    props: Omit<PropsOfElement<TExtendedElementType>, 'as' | keyof TProps> & { as?: TExtendedElementType } & TProps,
  ): JSX.Element | null;
  displayName?: string;

  defaultProps?: Partial<TProps & { as: TElementType }>;
  propTypes?: React.WeakValidationMap<TProps> & {
    // tslint:disable-next-line:no-any
    as: React.Requireable<string | ((props: any, context?: any) => any) | (new (props: any, context?: any) => any)>;
  };
}

//
// Compose types
//

export type ComposedComponent<TProps = {}> = React.FunctionComponent<TProps> & {
  fluentComposeConfig: ComposePreparedOptions;
};

export type InputComposeComponent<TProps = {}> = React.FunctionComponent<TProps> & {
  fluentComposeConfig?: ComposePreparedOptions;
};

export type Input<TElementType extends React.ElementType = 'div', TProps = {}> =
  | InputComposeComponent<TProps>
  | ComposeRenderFunction<TElementType, TProps & { as?: React.ElementType }>;

export type ComposeRenderFunction<TElementType extends React.ElementType = 'div', TProps = {}> = (
  props: TProps,
  ref: React.Ref<TElementType extends keyof HTMLElementTagNameMap ? HTMLElementTagNameMap[TElementType] : TElementType>,
  composeOptions: ComposePreparedOptions,
) => React.ReactElement | null;

export type ComposeOptions<TInputProps = {}, TInputStylesProps = {}, TParentProps = {}, TParentStylesProps = {}> = {
  className?: string;

  classes?: ClassDictionary | ClassFunction;

  displayName?: string;

  mapPropsToStylesProps?: (props: TParentStylesProps & TInputProps) => TInputStylesProps;

  handledProps?: (keyof TInputProps | 'as')[];

  overrideStyles?: boolean;

  slots?: Record<string, React.ElementType>;

  slotProps?: (props: TParentProps & TInputProps) => Record<string, object>;

  shorthandConfig?: ShorthandConfig<TParentProps & TInputProps>;
};

/**
 * Generic name to any dictionary.
 */
// tslint:disable-next-line:no-any
export type GenericDictionary = Record<string, any>;

/**
 * Generic set of module to class name map.
 */
export type ClassDictionary = Record<string, string>;

/**
 * Generic class resolver function type.
 */
export type ClassFunction = (state: GenericDictionary, slots: GenericDictionary) => ClassDictionary;

export type ComposePreparedOptions<TProps = {}, TState = TProps> = {
  className: string;
  classes: (undefined | ClassDictionary | ClassFunction)[];

  displayName: string;
  displayNames: string[];

  mapPropsToStylesPropsChain: ((props: object) => object)[];
  render: ComposeRenderFunction;

  handledProps: (keyof TProps)[];

  overrideStyles: boolean;

  slots: Record<string, React.ElementType> & { __self: React.ElementType };
  slotProps: ((props: TProps) => Record<string, object>)[];

  resolveSlotProps: <TResolvedProps>(props: TResolvedProps) => Record<string, object>;
  shorthandConfig: ShorthandConfig<TProps>;
};

/**
 * Options for classes resolver created using `createClassResolver`.
 */
export interface ResolveClassesOptions {
  state: GenericDictionary;
  slots: GenericDictionary;
  dynamicClasses?: (ClassDictionary | ClassFunction)[];
}
