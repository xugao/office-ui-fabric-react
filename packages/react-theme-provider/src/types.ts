import * as React from 'react';
import { IStyleFunctionOrObject } from '@uifabric/utilities';
import { ITheme } from '@uifabric/styling';

/**
 * A baseline set of color plates.
 */
export type ColorTokens = Partial<{
  background: string;
  contentColor: string;
  subTextColor: string;
  linkColor: string;
  iconColor: string;
  borderColor: string;
  dividerColor: string;
  focusColor: string;
  focusInnerColor: string;
}>;

/**
 * A set of states for each color plate to use.
 *
 * Note:
 *
 * State names here align to a consistent naming convention:
 *
 * The component is _____
 *
 * Bad: "hover", Good: "hovered"
 *
 * Additional considerations:
 *
 * The term "active" in css means that the keyboard or mouse button
 * which activates the component is pressed down. It is however ambiguous
 * with a focused state, as the HTML object model refers to the focused
 * element as the "activeElement". To resolve ambiguity and to be more
 * compatible with other platforms reusing token names, we have decided to snap
 * to "pressed".
 */
export type ColorTokenStates = Partial<{
  hovered: ColorTokens;
  pressed: ColorTokens;
  disabled: ColorTokens;
  checked: ColorTokens;
  checkedHovered: ColorTokens;
  checkedPressed: ColorTokens;
}>;

export type FontTokens = Partial<{
  fontFamily: string;
  fontSize: string;
  fontWeight: string;
}>;

export type ColorPlateSet = ColorTokens & ColorTokenStates;

/**
 * A token set can provide a single string or object, mapping additional sub-parts of a token set.
 */
export type TokenSetType = string | { [key: string]: TokenSetType | undefined };

/**
 * Recursive partial type.
 */
export type RecursivePartial<T> = {
  [P in keyof T]?: T[P] extends (infer U)[]
    ? RecursivePartial<U>[]
    : T[P] extends object
    ? RecursivePartial<T[P]>
    : T[P];
};

/**
 * A prepared (fully expanded) theme object.
 */
// export interface Theme {
//   tokens: {
//     body: ColorPlateSet & TokenSetType;
//     [key: string]: TokenSetType;
//   };

//   stylesheets: string[];
// }

export type Tokens = {
  body: ColorPlateSet & TokenSetType;
  [key: string]: TokenSetType;
};

export interface Theme extends ITheme {
  components?: {
    [componentName: string]: {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      styles?: IStyleFunctionOrObject<any, any>;
    };
  };
}

/**
 * A partial theme, provided by the customer. The internal `createTheme` helper will fill in the rest.
 */
export interface PartialTheme extends RecursivePartial<Theme> {}

/**
 * Typing containing the definition for the `style` and `tokens` props that will be extended for the calculation of the
 * style prop.
 */
export interface StyleProps<TTokens extends ColorPlateSet = ColorPlateSet> {
  style?: React.CSSProperties;
  tokens?: TTokens;
}

export interface StyleOptions<TProps> {
  slotProps: ((props: TProps) => Record<string, object>)[];
}
