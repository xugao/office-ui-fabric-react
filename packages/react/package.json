{
  "name": "@fluentui/react",
  "version": "8.0.0-beta.1",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-api": "just-scripts update-api",
    "update-snapshots": "just-scripts jest -u"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0-beta.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/jest": "~24.9.0",
    "@types/node": "^10.3.2",
    "@types/react": "16.8.25",
    "@types/react-dom": "16.8.4",
    "@types/react-test-renderer": "^16.0.0",
    "@types/sinon": "2.2.2",
    "@types/webpack-env": "1.15.1",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0-beta.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0-beta.0",
    "@fluentui/test-utilities": "^8.0.0-beta.0",
    "@fluentui/webpack-utilities": "^8.0.0-beta.0",
    "@fluentui/react-conformance": "^1.0.0",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0",
    "sinon": "^4.1.3"
  },
  "dependencies": {
    "@fluentui/font-icons-mdl2": "^8.0.0-beta.0",
    "@fluentui/react-button": "^1.0.0-beta.1",
    "@fluentui/react-checkbox": "^1.0.0-beta.1",
    "@fluentui/react-date-time": "^8.0.0-beta.1",
    "@fluentui/react-hooks": "^8.0.0-beta.0",
    "@fluentui/react-internal": "^8.0.0-beta.1",
    "@fluentui/react-link": "^1.0.0-beta.1",
    "@fluentui/react-slider": "^1.0.0-beta.1",
    "@fluentui/react-tabs": "^1.0.0-beta.1",
    "@fluentui/react-toggle": "^1.0.0-beta.1",
    "@fluentui/set-version": "^8.0.0-beta.0",
    "@fluentui/utilities": "^8.0.0-beta.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <17.0.0",
    "@types/react-dom": ">=16.8.0 <17.0.0",
    "react": ">=16.8.0 <17.0.0",
    "react-dom": ">=16.8.0 <17.0.0"
  }
}
