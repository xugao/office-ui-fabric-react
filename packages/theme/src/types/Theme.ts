import { IPartialTheme, ITheme } from './ITheme';
import { IStyleFunctionOrObject } from '@uifabric/utilities';

/**
 * A ramp of size values.
 */
export type SizeValue = 'smallest' | 'smaller' | 'small' | 'medium' | 'large' | 'larger' | 'largest';

/**
 * A baseline set of color plates.
 */
export type ColorTokenSet = {
  background?: string;
  contentColor?: string;
  secondaryContentColor?: string;
  linkColor?: string;
  iconColor?: string;
  borderColor?: string;
  dividerColor?: string;
  focusColor?: string;
  focusInnerColor?: string;
  opacity?: string;
};

/**
 * A set of states for each color plate to use.
 *
 * Note:
 *
 * State names here align to a consistent naming convention:
 *
 * The component is _____
 *
 * Bad: "hover", Good: "hovered"
 *
 * Additional considerations:
 *
 * The term "active" in css means that the keyboard or mouse button
 * which activates the component is pressed down. It is however ambiguous
 * with a focused state, as the HTML object model refers to the focused
 * element as the "activeElement". To resolve ambiguity and to be more
 * compatible with other platforms reusing token names, we have decided to snap
 * to "pressed".
 */
export type ColorTokens = ColorTokenSet & {
  hovered?: ColorTokens;
  pressed?: ColorTokens;
  disabled?: ColorTokens;
  checked?: ColorTokens;
  checkedHovered?: ColorTokens;
  checkedPressed?: ColorTokens;
};

export type FontTokens = Partial<{
  fontFamily: string;
  fontSize: string;
  fontWeight: string;
}>;

/**
 * A token set can provide a single string or object, mapping additional sub-parts of a token set.
 */
export type TokenSetType = { [key: string]: TokenSetType | string | number | undefined };

/**
 * Recursive partial type.
 */
export type RecursivePartial<T> = {
  [P in keyof T]?: T[P] extends Array<infer I> ? Array<RecursivePartial<I>> : RecursivePartial<T[P]>;
};

export interface Tokens {
  color: {
    body: ColorTokenSet & TokenSetType;
    brand: ColorTokenSet & TokenSetType;
    [key: string]: TokenSetType;
  };

  [key: string]: TokenSetType;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export declare type Variants = Record<string, any>;

export interface ComponentStyles {
  [componentName: string]: {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    styles?: IStyleFunctionOrObject<any, any>;
    variants?: Variants;
  };
}

/**
 * A prepared (fully expanded) theme object.
 */
export interface Theme extends ITheme {
  components?: ComponentStyles;
  stylesheets?: string[];
  id?: string;

  /** @internal
   * This is currently only for internal use and not production-ready.
   * */
  tokens?: RecursivePartial<Tokens>;
}

/**
 * A partial theme object.
 */
export interface PartialTheme extends IPartialTheme {
  components?: ComponentStyles;
  tokens?: RecursivePartial<Tokens>;
  stylesheets?: string[];
}
