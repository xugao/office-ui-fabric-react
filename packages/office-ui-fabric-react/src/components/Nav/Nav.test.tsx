import * as React from 'react';

import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';

import { Nav } from './Nav';
import { NavBase } from './Nav.base';
import { INavLink } from './Nav.types';

const linkOne: INavLink = {
  key: 'Bing',
  name: 'Bing',
  url: 'http://localhost/#/testing1'
};

const linkTwo: INavLink = {
  key: 'OneDrive',
  name: 'OneDrive',
  url: 'http://localhost/#/testing2'
};

describe('Nav', () => {
  it('renders Nav correctly', () => {
    const component = renderer.create(
      <Nav
        groups={[
          {
            links: [
              {
                name: '',
                url: ''
              }
            ]
          }
        ]}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick() correctly', () => {
    const handler = jest.fn();
    const nav = mount<NavBase>(
      <NavBase
        groups={[
          {
            links: [
              {
                name: 'foo',
                url: 'http://example.com',
                onClick: handler
              }
            ]
          }
        ]}
      />
    );

    const link = nav.find('a.ms-Button');
    link.simulate('click');
    expect(handler.mock.calls.length).toBe(1);
  });

  it('do not call onClick() on disabled link', () => {
    const handler = jest.fn();
    const nav = mount<NavBase>(
      <NavBase
        groups={[
          {
            links: [
              {
                name: 'foo',
                url: 'http://example.com',
                onClick: handler,
                disabled: true
              }
            ]
          }
        ]}
      />
    );

    const link = nav.find('button.ms-Button');
    link.simulate('click');
    expect(handler.mock.calls.length).toBe(0);
  });

  it('sets ARIA label on the nav element', () => {
    const label = 'The navigation label';
    const nav = mount<NavBase>(<Nav ariaLabel={label} groups={[]} />);
    expect(nav.find('[role="navigation"]').prop('aria-label')).toEqual(label);
  });

  it('uses location.href to determine link selected status if state/props is not set', () => {
    const nav = mount<NavBase>(
      <Nav
        groups={[
          {
            links: [linkOne, linkTwo]
          }
        ]}
      />
    );
    window.history.pushState({}, '', '/#/testing1');
    nav.instance().forceUpdate();

    expect(nav.getDOMNode().querySelectorAll('.ms-Nav-compositeLink.is-selected').length).toBe(1);
    expect(nav.getDOMNode().querySelectorAll('.ms-Nav-compositeLink.is-selected')[0].textContent).toEqual(linkOne.name);
  });

  it('prioritizes state over location.href to determine link selected status', () => {
    const nav = mount<NavBase>(
      <Nav
        groups={[
          {
            links: [linkOne, linkTwo]
          }
        ]}
      />
    );
    window.history.pushState({}, '', '/#/testing2');
    nav.instance().forceUpdate();

    nav
      .find('.ms-Button')
      .first()
      .simulate('click');
    expect(nav.getDOMNode().querySelectorAll('.ms-Nav-compositeLink.is-selected').length).toBe(1);
    expect(nav.getDOMNode().querySelectorAll('.ms-Nav-compositeLink.is-selected')[0].textContent).toEqual(linkOne.name);
  });

  it('places the correct values on rel depending on the url and target specified', () => {
    const linkWithNoTargetSpecified: INavLink = {
      key: 'Link1',
      name: 'Link1',
      url: 'https://cdpn.io/bar'
    };
    const linkWithRelativeURL: INavLink = {
      key: 'Link2',
      name: 'Link2',
      target: '_blank',
      url: '/foo'
    };
    const linkWithNonRelativeURL: INavLink = {
      key: 'Link3',
      name: 'Link3',
      target: '_blank',
      url: 'https://cdpn.io/bar'
    };
    const linkWithNonRelativeURL2: INavLink = {
      key: 'Link4',
      name: 'Link4',
      target: '_blank',
      url: 'http://cdpn.io/bar'
    };
    const nav = mount<NavBase>(
      <Nav groups={[{ links: [linkWithNoTargetSpecified, linkWithRelativeURL, linkWithNonRelativeURL, linkWithNonRelativeURL2] }]} />
    );

    const links = nav.getDOMNode().querySelectorAll('.ms-Nav-link');
    expect(links.length).toBe(4);
    expect(links[0].getAttribute('rel')).toBeFalsy();
    expect(links[1].getAttribute('rel')).toBeFalsy();
    expect(links[2].getAttribute('rel')).toBe('noopener noreferrer');
    expect(links[3].getAttribute('rel')).toBe('noopener noreferrer');
  });
});
