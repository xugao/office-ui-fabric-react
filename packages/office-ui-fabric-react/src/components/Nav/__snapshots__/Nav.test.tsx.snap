// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders Nav correctly 1`] = `
<div
  className=
      ms-FocusZone
      &:focus {
        outline: none;
      }
  data-focuszone-id="FocusZone0"
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDownCapture={[Function]}
>
  <nav
    className=
        ms-Nav
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          -webkit-overflow-scrolling: touch;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow-y: auto;
          user-select: none;
        }
    role="navigation"
  >
    <div
      className="ms-Nav-group is-expanded"
    >
      <div
        className=
            ms-Nav-groupContent
            {
              animation-duration: 0.367s;
              animation-fill-mode: both;
              animation-name: keyframes from{opacity:0;}to{opacity:1;} keyframes from{transform:translate3d(0,-20px,0);}to{transform:translate3d(0,0,0);};
              animation-timing-function: cubic-bezier(.1,.9,.2,1);
              display: block;
              margin-bottom: 40px;
            }
      >
        <ul
          className=
              ms-Nav-navItems
              {
                list-style-type: none;
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
                padding-bottom: 0px;
                padding-left: 0px;
                padding-right: 0px;
                padding-top: 0px;
              }
          role="list"
        >
          <li
            className=
                ms-Nav-navItem
                {
                  padding-bottom: 0px;
                  padding-left: 0px;
                  padding-right: 0px;
                  padding-top: 0px;
                }
            role="listitem"
          >
            <div
              className=
                  ms-Nav-compositeLink
                  {
                    color: #323130;
                    display: block;
                    position: relative;
                  }
              name=""
            >
              <button
                className=
                    ms-Button
                    ms-Button--action
                    ms-Button--command
                    ms-Nav-link
                    {
                      -moz-osx-font-smoothing: grayscale;
                      -webkit-font-smoothing: antialiased;
                      background-color: transparent;
                      border-radius: 2px;
                      border: 1px solid transparent;
                      box-sizing: border-box;
                      color: #323130;
                      cursor: pointer;
                      display: block;
                      font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                      font-size: 14px;
                      font-weight: 400;
                      height: 44px;
                      line-height: 44px;
                      outline: transparent;
                      overflow: hidden;
                      padding-bottom: 0;
                      padding-left: 27px;
                      padding-right: 20px;
                      padding-top: 0;
                      position: relative;
                      text-align: center;
                      text-decoration: none;
                      text-overflow: ellipsis;
                      user-select: none;
                      white-space: nowrap;
                      width: 100%;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid #ffffff;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
                      border: none;
                      bottom: -2px;
                      left: -2px;
                      outline-color: ButtonText;
                      right: -2px;
                      top: -2px;
                    }
                    &:active > * {
                      left: 0px;
                      position: relative;
                      top: 0px;
                    }
                    &:hover {
                      color: #0078d4;
                    }
                    @media screen and (-ms-high-contrast: active){&:hover {
                      border-color: Highlight;
                      color: Highlight;
                    }
                    &:hover .ms-Button-icon {
                      color: #0078d4;
                    }
                    &:active {
                      color: #000000;
                    }
                    &:active .ms-Button-icon {
                      color: #004578;
                    }
                    @media screen and (-ms-high-contrast: active){& {
                      border-color: transparent;
                    }
                    @media screen and (-ms-high-contrast: active){&:focus {
                      border-color: WindowText;
                    }
                    .ms-Nav-compositeLink:hover & {
                      background-color: #f3f2f1;
                    }
                data-is-focusable={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                title=""
                type="button"
              >
                <span
                  className=
                      ms-Button-flexContainer
                      {
                        align-items: center;
                        display: flex;
                        flex-wrap: nowrap;
                        height: 100%;
                        justify-content: flex-start;
                      }
                  data-automationid="splitbuttonprimary"
                >
                  <div
                    className=
                        ms-Nav-linkText
                        {
                          margin-bottom: 0;
                          margin-left: 4px;
                          margin-right: 4px;
                          margin-top: 0;
                          overflow: hidden;
                          text-align: left;
                          text-overflow: ellipsis;
                          vertical-align: middle;
                        }
                  >
                    
                  </div>
                </span>
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
